### DISCUSSION TITLE
Where should `evaluate` live?

### DISCUSSION THREAD

**Discussion by ocharles** [haskell-effectful/effectful#248] [Q&A] [answered]
I find myself often wanting or needing `evaluate`, but the one from `base` requires `liftIO`. This seems like a shame, given that we have `MonadThrow (Eff es)`. Should `effectful` provide it's own `evaluate`? Should it be added to `MonadThrow`?

**Comment by arybczak** [comment]
Having `evaluateEff :: a -> Eff es a` (given that instances of classes from `exceptions` are always available) sounds reasonable to me.

**Comment by arybczak** [comment]
Actually, I think I'll add `Effectful.Exception` module to `effectful-core` that just exports `Control.Exception.Safe` from`safe-exceptions` and then `evaluate` and `evaluateDeep` can be added there. It'll also improve discoverability, since some people miss that `effectful` works out of the box with `MonadThrow`/`MonadCatch`/`MonadMask`.

It's tempting to export `Control.Monad.Catch` from `exceptions` instead to avoid the extra dependency, but over the years I realized that `safe-exceptions` makes it way easier to write code that handles async exceptions properly (especially for people who are not experts in the matter) and ultimately this is more important.

**Reply by ocharles** [reply]
That sounds good to me. Would you also specialize all of this to `Eff`?

**Reply by arybczak** [reply]
No, simply re-export the module since it works out of the box. I like avoiding unnecessary work ;)

**Reply by arybczak** [reply, marked as answer]
Done in #252.
